src/Engine/ProtoAssociative/CodeGen.cs

```diff
@@ -3776,13 +3792,25 @@ private void EmitFunctionDefinitionNode(AssociativeNode node, ref ProtoCore.Type
                 // Get the exisitng procedure that was added on the previous pass
                 if (ProtoCore.DSASM.Constants.kInvalidIndex == globalClassIndex)
                 {
-                    var procNode = codeBlock.procedureTable.GetFunctionBySignature(funcDef.Name, argList);
+                    codeBlock.procedureTable.GetFunctionBySignature(new ProcedureMatchOptions() 
+                    { 
+                        FunctionName = funcDef.Name, 
+                        ParameterTypes = argList,
+                        IsStatic = funcDef.IsStatic 
+                    }, out ProcedureNode procNode);
+
                     globalProcIndex = procNode == null ? Constants.kInvalidIndex : procNode.ID;
                     localProcedure = codeBlock.procedureTable.Procedures[globalProcIndex];
                 }
                 else
                 {
-                    var procNode = core.ClassTable.ClassNodes[globalClassIndex].ProcTable.GetFunctionBySignature(funcDef.Name, argList);
+                    core.ClassTable.ClassNodes[globalClassIndex].ProcTable.GetFunctionBySignature(new ProcedureMatchOptions()
+                    {
+                        FunctionName = funcDef.Name,
+                        ParameterTypes = argList,
+                        IsStatic = funcDef.IsStatic
+                    }, out ProcedureNode procNode);
+
                     globalProcIndex = procNode == null ? Constants.kInvalidIndex : procNode.ID;
```

Comment:
If `globalProcIndex` is equal to `kInvalidIndex` (-1), the next line will bomb so I think we should just have a `Validity.Assert` here saying that `procNode` cannot be `null` at this point.

Reply:
I know this is existing code, not yours but just an improvement suggestion.

Reply:
nice, good catch!